processing/src/main/java/org/apache/carbondata/processing/merger/CarbonDataMergerUtil.java

```diff
@@ -351,6 +357,27 @@ public static boolean updateLoadMetadataWithMergeStatus(List<LoadMetadataDetails
           loadMetadataDetails.setMajorCompacted("true");
         }
 
+        if (DataMapUtil.isMVdatamapTable(carbonTable)) {
+          // If table is mv datamap table, then get segment mapping and set to extraInfo
+          DataMapSchema dataMapSchema = null;
+          try {
+            dataMapSchema = DataMapStatusManager.getDataMapSchema(carbonTable.getTableName()
+                .substring(0,
```

Comment:
I feel it is not a good way to get the datamap name like this, better add the required information to tableproperty and get them